watertap/unit_models/translators/translator_asm2d_adm1.py

```diff
@@ -0,0 +1,1246 @@
+        def Ssu_mapping(blk, t):
+            return Expr_if(
+                blk.SN_org[t] >= blk.properties_in[t].conc_mass_comp["S_F"],
+                eps * pyunits.kg / pyunits.m**3,
+                blk.properties_in[t].conc_mass_comp["S_F"] - blk.SN_org[t],
+            )
+
+        @self.Expression(self.flowsheet().time, doc="Amino acids mapping")
+        def Saa_mapping(blk, t):
+            return Expr_if(
+                blk.SN_org[t] >= blk.properties_in[t].conc_mass_comp["S_F"],
+                blk.properties_in[t].conc_mass_comp["S_F"],
+                blk.SN_org[t],
+            )
+
+        @self.Constraint(
+            self.flowsheet().time,
+            doc="Monosaccharides output",
+        )
+        def Ssu_output(blk, t):
+            return blk.properties_out[t].conc_mass_comp["S_su"] == blk.Ssu_mapping[t]
+
+        @self.Constraint(
+            self.flowsheet().time,
+            doc="Amino acids output",
+        )
+        def Saa_output(blk, t):
+            return blk.properties_out[t].conc_mass_comp["S_aa"] == blk.Saa_mapping[t]
+
+        @self.Expression(self.flowsheet().time, doc="S_F concentration at step 3")
+        def SF_AS3(blk, t):
+            return (
+                blk.properties_in[t].conc_mass_comp["S_F"]
+                - blk.Ssu_mapping[t]
+                - blk.Saa_mapping[t]
+            )
+
+        @self.Expression(self.flowsheet().time, doc="S_NH4 concentration at step 3")
+        def SNH4_AS3(blk, t):
+            return (
+                blk.SNH4_AS2[t]
+                + blk.properties_in[t].conc_mass_comp["S_F"]
+                * blk.config.inlet_reaction_package.i_NSF
+                - blk.Saa_mapping[t]
+                * blk.config.outlet_reaction_package.Ni["S_aa"]
+                * mw_n
+            )
+
+        @self.Expression(self.flowsheet().time, doc="S_PO4 concentration at step 3")
+        def SPO4_AS3(blk, t):
+            return (
+                blk.SPO4_AS2[t]
+                + blk.properties_in[t].conc_mass_comp["S_F"]
+                * blk.config.inlet_reaction_package.i_PSF
+            )
+
+        @self.Expression(self.flowsheet().time, doc="S_IC concentration at step 3")
+        def SIC_AS3(blk, t):
+            return (
+                blk.SIC_AS2[t]
+                + blk.properties_in[t].conc_mass_comp["S_F"]
+                * blk.config.inlet_reaction_package.i_CSF
+                - blk.Ssu_mapping[t]
+                * blk.config.outlet_reaction_package.Ci["S_su"]
+                * mw_c
+                - blk.Saa_mapping[t]
+                * blk.config.outlet_reaction_package.Ci["S_aa"]
+                * mw_c
+            )
+
+        # -----------------------------------------DecaySwitch.on----------------------------------------------------------#
+
+        # -------------------------------------------Step 4----------------------------------------------------------------#
```

Comment:
Now, I see the difference of DecaySwitch options. Do you know why don't we want to directly add a config in this translator unit block instead of in properties package?

Reply:
This ties a bit to your previous comment...we could do this, but I assume it's better to have it in one place and import from there rather than recreate it in the translator block.

Reply:
I don't mean we have the config both on properties model and the translator, it seems to me that the decayswitch is only related to the translator block, then probably we could consider add it as a config only for this translator, not for the properties model.

Reply:
The DecaySwitch is used a decent bit in the `modified_asm2d_reactions.py` file though. For example, Line 1298 is the first use of it in this file. So it's used in the reaction file and in the translator unit model file.

Reply:
OK. I understand, make sense to me!